lib/cmsis_rtos_v1: Check for mutex owner in cmsis

rgundi opened the pull request:
Check for the mutex owner in the cmsis layer itself while releasing.
Fixes #9574.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

rgundi committed:
lib/cmsis_rtos_v1: Check for mutex owner in cmsis layer

Check for the mutex owner in the cmsis layer itself while releasing.
Fixes #9574.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

pswarnak commented on tests/cmsis_rtos_v1/src/mutex.c:
```diff
@@ -101,6 +101,9 @@ void tThread_entry_lock_timeout(void const *arg)
 	status = osMutexWait((osMutexId)arg, TIMEOUT - 100);
 	zassert_true(status == osErrorTimeoutResource, NULL);
 
+	status = osMutexRelease((osMutexId)arg);
```
can we please add a comment mentioning, what the code snippet is validating ?
something like "Mutex Ownership check - try to release a mutex, acquired by another thread"

rgundi replied:
Addressed.

pswarnak reviewed (commented):
I know the changes are really minor, but just curious to know, if we follow commit rule for lib too, similar to kernel.
That means, Shall we create two different commits for each 1 liner changes in lib and tests OR just one commit is acceptable for minor changes.

rgundi committed:
tests/cmsis_rtos_v1: Test mutex ownership during releasing

Introduce test to check that only the thread which locks the mutex
shall be able to release it.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

rgundi commented:
@pswarnak : Addressed your comments.
